lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -800,19 +795,26 @@ class ClassMetadataInfo implements ClassMetadata
     /** @var InstantiatorInterface|null */
     private $instantiator;
 
+    /** @var TypedFieldMapper $typedFieldMapper */
+    private $typedFieldMapper;
+
     /**
      * Initializes a new ClassMetadata instance that will hold the object-relational mapping
      * metadata of the class with the given name.
      *
      * @param string $entityName The name of the entity class the new instance is used for.
      * @psalm-param class-string<T> $entityName
      */
-    public function __construct($entityName, ?NamingStrategy $namingStrategy = null)
+    public function __construct($entityName, ?NamingStrategy $namingStrategy = null, ?TypedFieldMapper $typedFieldMapper = null)
     {
         $this->name           = $entityName;
         $this->rootEntityName = $entityName;
         $this->namingStrategy = $namingStrategy ?: new DefaultNamingStrategy();
         $this->instantiator   = new Instantiator();
+
+        if (PHP_VERSION_ID >= 70400) {
+            $this->typedFieldMapper = $typedFieldMapper ?: new DefaultTypedFieldMapper();
```

Comment:
```suggestion
            $this->typedFieldMapper = $typedFieldMapper ?? new DefaultTypedFieldMapper();
```

Reply:
thisis used a bit above also

```php
        $this->namingStrategy = $namingStrategy ?: new DefaultNamingStrategy();
```

Do you want me to change that too?

Reply:
Yes please.

Reply:
done already, all feedback incorporated